src/ServiceControl/MessageFailures/Handlers/UnArchiveMessagesHandler.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace ServiceControl.MessageFailures.Handlers
+{
+    using System.Linq;
+    using Contracts.MessageFailures;
+    using InternalMessages;
+    using NServiceBus;
+    using Raven.Client;
+
+    public class UnArchiveMessagesHandler : IHandleMessages<UnArchiveMessages>
+    {
+        public IDocumentSession Session { get; set; }
```

Comment:
> As we aren't executing in a transaction

we are in a transaction, MSMQ transaction, will roll back and try again.
Or maybe I am misunderstanding your comment ?